### ISSUE TITLE
Timeseries Chart Scale skips to the next month showing confusing results

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#49662]
### Describe the bug

Easier to see than to explain. This is the data as a Table:

![Image](https://github.com/user-attachments/assets/eddc5475-d406-485a-a1a0-5b441ad70990)

This is the Data as a bar chart with Ordinal Scale:

![Image](https://github.com/user-attachments/assets/f1913fbf-8c5a-4959-b24e-7740d100e4ce)

This is the Data as a bar chart with Timeseries Scale:

![Image](https://github.com/user-attachments/assets/a2f61595-dd94-4860-8ba7-fbe91b9f55d6)

### To Reproduce

I am using Postgres but you can checkout this [question on stats](https://stats.metabase.com/question/20453-timeseries-confusing-chart)
 
1. Go to New -> SQL Question -> Run the below:

```
Select TO_DATE('30/06/2015', 'DD/MM/YYYY') AS date, 5555 AS value
UNION
Select TO_DATE('30/09/2015', 'DD/MM/YYYY') AS date, 6555 AS value
```

2. Visualise as a Bar chart and set the Scale to a Timeseries, notice the actual Data doesn't match with the chart:

![Image](https://github.com/user-attachments/assets/e3bd3aef-9e9f-410b-b8e0-b478a6da1cfa)

![Image](https://github.com/user-attachments/assets/d673ef4f-ec40-4c88-8fb0-30f0591ed9ff)

### Expected behavior

The Timeseries scale matches what you get in the chart. So my expectation would be to get a June Bar and a July, August empty bars then a September Bar. Something like the below when you change the Date from 30th to 14th

![Image](https://github.com/user-attachments/assets/668396cc-798c-4a46-8f0a-b69c37664ecd)

### Logs

None that are relevant

### Information about your Metabase installation

Happens both on 50 and 51

### Severity

Might look like a cosmetic bug but when you hit this you don't understand what is going on

**Comment by lorem--ipsum** [comment]
I think what happens is the time axis displays ticks at month starts, which is what usually makes more sense. The data dots look like they are on the wrong month because they're so close to the end of the month before. The tooltip displays the exact data though, and it's not displaying the full date which might desrve its own investigation. With more data points, the chart looks a lot less confusing. This query for instance:
```
SELECT 
  TO_DATE(date_str, 'DD/MM/YYYY') AS date,
  value
FROM (
  VALUES
    ('15/01/2015', 4200),
    ('28/01/2015', 4350),
    ('17/02/2015', 4500),
    ('10/03/2015', 5100),
    ('25/03/2015', 5200),
    ('08/04/2015', 5450),
    ('20/05/2015', 5300),
    ('12/06/2015', 5600),
    ('28/06/2015', 5800),
    ('15/07/2015', 6200),
    ('20/08/2015', 5900),
    ('05/09/2015', 5500),
    ('25/09/2015', 5400),
    ('10/10/2015', 5100),
    ('08/11/2015', 4800),
    ('22/11/2015', 4900),
    ('15/12/2015', 5200),
    ('10/01/2016', 5300),
    ('22/02/2016', 5600),
    ('05/03/2016', 5800),
    ('20/03/2016', 6000),
    ('12/04/2016', 6300),
    ('25/05/2016', 6100),
    ('08/06/2016', 6400),
    ('15/07/2016', 6800),
    ('30/07/2016', 6700),
    ('18/08/2016', 6500),
    ('05/09/2016', 6200),
    ('22/10/2016', 5900),
    ('10/11/2016', 5700),
    ('28/11/2016', 5500),
    ('15/12/2016', 5800)
) AS date_values(date_str, value);
```
produces the following result:

<img width="1488" alt="Image" src="https://github.com/user-attachments/assets/9521d0d6-e89b-4add-a6ad-7186b96b7cf7" />

In summary, I'm 90% sure that this behavior is correct here, although I'm open to be convinced otherwise :-)

**Comment by stefano337** [comment]
I'm experiencing this issue.

As you can see, I'm reporting on some rows that have a date in May. I'm specifically filtering out dates in April.

Problem 1: While the bar does show up above May 2025, there's still a label in the x-axis that shows April 2025.
Problem 2: The tooltip shows the wrong month (April 2025) on hover.

This is driving internal confusion about data.

![Image](https://github.com/user-attachments/assets/996a4398-2751-41da-a384-7ae1bae793cb)

**Comment by stefano337** [comment]
When I change the scale from Timeseries to Ordinal, it gets even worse, showing all of the data under April when the actual datapoints are in May:

![Image](https://github.com/user-attachments/assets/269372eb-400a-4e0c-a7a0-9d27ad5e2a78)

**Comment by stefano337** [comment]
Here's a sample of the data so you can see they're all in May:

![Image](https://github.com/user-attachments/assets/f6601362-6e42-4889-bc00-c26b0ee042e1)
